src/services/etherspot.js

```diff
@@ -278,16 +278,36 @@ export class EtherspotService {
     });
   }
 
-  clearTransactionsBatch() {
-    return this.sdk.clearGatewayBatch();
+  clearTransactionsBatch(chain: Chain): void {
+    const sdk = this.getSdkForChain(chain);
+
+    if (!sdk) {
+      reportErrorLog('clearTransactionsBatch failed: no SDK for chain set', { chain });
+      throw new Error(t('error.unableToResetTransactions'));
+    }
+
+    sdk.clearGatewayBatch();
   }
 
-  setTransactionsBatch(transactions: EthereumTransaction[]) {
-    return Promise.all(transactions.map((transaction) => this.sdk.batchExecuteAccountTransaction(transaction)));
+  setTransactionsBatch(transactions: EthereumTransaction[], chain: Chain) {
```

Comment:
Consider moving `chain` param in etherspotService methods to the front, as this is kind of scope for the whole operation

```suggestion
  setTransactionsBatch(chain: Chain, transactions: EthereumTransaction[]) {
```